tests/test_config_manager.py

```
@@ -196,3 +196,305 @@ def test_none_overrides_not_serialized(self, tmp_path: Path) -> None:
+            migrations_mod.MIGRATIONS.clear()
+            migrations_mod.MIGRATIONS.update(original_migrations)
+
+    def test_migration_failure_falls_back_to_defaults(
+        self, tmp_path: Path, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        """F6: if a registered migration raises, the load path must
+        log the failure and fall back to AppConfig defaults rather
+        than crashing the CLI at startup."""
+        from config import migrations as migrations_mod
+
+        def broken_migration(data: dict) -> dict:
```

Ruff (0.15.11)

[warning] Unused function argument: `data`

(ARG001)